pypit/masterframe.py

```diff
@@ -0,0 +1,109 @@
+""" Abstract class for Master Image frames
+      This could go inside armasters.py
+"""
+from __future__ import absolute_import, division, print_function
+
+import warnings
+
+from pypit import msgs
+from pypit import ardebug as debugger
+from pypit import armasters
+
+from abc import ABCMeta
+
+default_settings = dict(masters={
+    'directory': 'MF/',  # local to the run directory
+    'reuse': False,
+    'force': False,
+    'loaded': [],
+    'setup': None,
+})
+
+
+class MasterFrame(object):
+    """
+    This class is designed to gather a set of like methods
+    for Master calibration frames
+
+    Parameters
+    ----------
+    frametype : str
+    setup : str
+    settings : dict
+
+    Attributes
+    ----------
+    """
+    __metaclass__ = ABCMeta
```

Comment:
Out of my depth again.  I need to read up on abc, but do you have a quick response for why this is needed?

Reply:
I'm not sure if it is needed and we can surely remove
it anytime.  I just think it is good practice when you never
intend to instantiate this object but only inherit it.